docs/user/alerting/alerting-troubleshooting.asciidoc

```diff
@@ -69,3 +69,168 @@ Configuration options are available to specialize connections to TLS servers,
 including ignoring server certificate validation, and providing certificate
 authority data to verify servers using custom certificates.  For more details, 
 see <<action-settings,Action settings>>.
+
+[float]
+[[rules-long-execution-time]]
+=== Identify long running rules
+
+The following query can help you identify rules that are taking a long time to execute and may be impacting the overall health of your deployment.
+
+Query for a list of rule ids, bucketed by their execution times:
+
+[source,sh]
+--------------------------------------------------
+$ GET /.kibana-event-log*/_search
+{
+  "size": 0,
+  "query": {
+    "bool": {
+      "filter": [
+        {
+          "range": {
+            "@timestamp": {
+              "gte": "now-1d", <1>
+              "lte": "now"
+            }
+          }
+        },
+        {
+          "term": {
+            "event.action": {
+              "value": "execute"
+            }
+          }
+        }
+      ]
+    }
+  },
+  "aggs": {
+    "ruleIdsByExecutionDuration": {
+      "histogram": {
+        "field": "event.duration",
+        "interval": 10000000000 <2>
```

Comment:
uggh; I wonder if there's anything to help here, to get these nano's scaled down to milli's.  I think I've seen suggestions that this field should be a multi-field, with a subfield for the millis, but I don't quite know how that would work.  Wonder how hard it would be to "runtime" a milli version of this field, for just this call?

Reply:
Good idea! Added runtime fields to the query in this commit https://github.com/elastic/kibana/pull/98773/commits/fbc650dc6b5ce5267eda7d118913fbe04d73a552